infrastructure/create-github-environment.js

```diff
@@ -0,0 +1,226 @@
+const sodium = require('libsodium-wrappers')
+const { Octokit } = require('@octokit/core')
+const { writeFileSync } = require('fs')
+
+const config = {
+  environment: '',
+  repo: {
+    REPOSITORY_ID: '',
+    DOCKERHUB_ACCOUNT: '', // This may be a dockerhub organisation or the same as the username
+    DOCKERHUB_REPO: '',
+    DOCKER_USERNAME: process.env.DOCKER_USERNAME,
+    DOCKER_TOKEN: process.env.DOCKER_TOKEN
+  },
+  ssh: {
+    KNOWN_HOSTS: process.env.KNOWN_HOSTS,
+    SSH_HOST: process.env.SSH_HOST, // IP address for the manager
+    SSH_USER: process.env.SSH_USER,
+    // SUDO_PASSWORD: process.env.SUDO_PASSWORD, // in case your user is not root
+    SSH_KEY: process.env.SSH_KEY // id_rsa
+  },
+  infrastructure: {
+    DISK_SPACE: '',
+    HOSTNAME: '', // server machine hostname used when provisioning - TODO: Adapt to support 3 or 5 replicas
+    DOMAIN: '', // web hostname applied after all public subdomains in Traefik,
+    REPLICAS: '1' // TODO: Adapt to support 3 or 5 replicas
+  },
+  services: {
+    SENTRY_DSN: process.env.SENTRY_DSN || '',
+    ELASTALERT_SLACK_WEBHOOK: process.env.ELASTALERT_SLACK_WEBHOOK || '',
+    INFOBIP_API_KEY: process.env.INFOBIP_API_KEY || '',
+    INFOBIP_GATEWAY_ENDPOINT: process.env.INFOBIP_GATEWAY_ENDPOINT || '',
+    INFOBIP_SENDER_ID: process.env.INFOBIP_SENDER_ID || '' // the name of the SMS sender e.g. OpenCRVS
+  },
+  seeding: {
+    ACTIVATE_USERS: true,
+    AUTH_HOST: '',
+    COUNTRY_CONFIG_HOST: '',
+    GATEWAY_HOST: ''
+  },
+  smtp: {
+    SMTP_HOST: process.env.SMTP_HOST || '',
+    SMTP_USERNAME: process.env.SMTP_USERNAME || '',
+    SMTP_PASSWORD: process.env.SMTP_PASSWORD || '',
+    EMAIL_API_KEY: process.env.EMAIL_API_KEY || '',
+    SMTP_PORT: '',
+    ALERT_EMAIL: ''
+  },
+  vpn: {
+    // openconnect details for optional VPN
+    VPN_PROTOCOL: '', // e,g, fortinet, wireguard etc
+    VPN_HOST: process.env.VPN_HOST || '',
+    VPN_PORT: process.env.VPN_PORT || '',
+    VPN_USER: process.env.VPN_USER || '',
+    VPN_PWD: process.env.VPN_PWD || '',
+    VPN_SERVERCERT: process.env.VPN_SERVERCERT || ''
+  },
+  whitelist: {
+    CONTENT_SECURITY_POLICY_WILDCARD: '', // e.g. *.<your-domain>
+    CLIENT_APP_URL: '',
+    LOGIN_URL: ''
+  },
+  backup: {
+    BACKUP_HOST: process.env.BACKUP_HOST || '',
+    BACKUP_DIRECTORY: '',
+    qa: {
+      RESTORE_DIRECTORY: '' // If making use of script to restore a production backup on QA for regular monitoring
+    }
+  }
+}
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN
```

Comment:
What permissions would the token need?

Reply:
@rikukissa is it "repo" & "codespaces"?  

![Screenshot 2023-12-07 at 17 52 55](https://github.com/opencrvs/opencrvs-farajaland/assets/3169954/3b7d3dec-22a3-4338-baa5-c47256787adc)
![Screenshot 2023-12-07 at 17 51 48](https://github.com/opencrvs/opencrvs-farajaland/assets/3169954/54a39230-0d6d-492e-9aee-64cf2e938755)

Reply:
Seems to be enough after testing.